graphql/resolvers/exerciseCrud.ts

```diff
@@ -75,26 +73,24 @@ export const updateExercise = async (
       module: true
     }
   })
-}
-
-export const deleteExercise = async (
-  _parent: void,
-  arg: MutationDeleteExerciseArgs,
-  { req }: Context
-): Promise<Exercise> => {
-  const { id } = arg
-  const authorId = req.user?.id
-  if (!authorId) throw new Error('No User')
-
+})
+
+export const deleteExercise = withUserContainer<
+  Promise<Exercise>,
+  MutationDeleteExerciseArgs
+>(async (_parent: void, args: MutationDeleteExerciseArgs, ctx: Context) => {
+  const { req } = ctx
+  const { id } = args
   const exercise = await prisma.exercise.findUnique({
     where: {
       id
     }
   })
 
-  if (!isAdmin(req) && exercise?.authorId !== authorId) {
+  const authorId = req.user?.id
+
+  if (!isAdmin(req) && exercise?.authorId !== authorId)
```

Comment:
`withAdminContainer` and then we don't need the `isAdmin` check

Reply:
@songz So in this example, I been getting stuck how to handle when you need to also check for the ` && exercise?.authorId !== authorId` on line 92 or something similar when the resolver needs to query something in prisma, for example lines 84-88:
```
 const exercise = await prisma.exercise.findUnique({
    where: {
      id
    }
  })
```
That part of the check can't be captured universally with the container, since it is unique to a specific resolver. 

The if statement on line 92 is also an `&&` operator. So checking the conditions individually with a separate if statement for each wont work.

Reply:
IMO it's fine and would be a good choice to keep `if (exercise?.authorId !== authorId)` in this file since its specific to this resolver.

Reply:
oh this is my bad, deleting exercise seems like it can be used for both admins **and** regular users, so `withAdminContainer` wouldn't be appropriate here.

Reply:
it would be more appropriate like `updateLesson` or `deleteModule` etc, which you could just do in a separate PR.

Reply:
@songz Hey no worries. Seems you need to either be an admin, or the regular user who created the exercise to be able to delete it, which makes sense. 

I applied some changes based on @anthonykhoa  comments and think it's good to merge now. Then I can make a separate PR for the AdminContainer